tslint.json

```diff
@@ -1,8 +1,9 @@
 {
   "extends": "tslint:recommended",
   "rules": {
+    "no-string-literal": true,
     "no-reference": false,
-    "no-floating-promises": true,
+    "no-floating-promises": false,
```

Comment:
Oh, interesting change! :)

If `true`:

```
ERROR: - Promises must be handled appropriately
src/renderer/components/catalog/PublicationAddButton.tsx:54:9
src/renderer/components/dialog/DeleteOpdsFeedConfirm.tsx:52:9
src/renderer/components/dialog/DeletePublicationConfirm.tsx:52:9
src/renderer/components/dialog/FileImport.tsx:40:13
src/renderer/components/dialog/LcpAuthentication.tsx:76:9
src/renderer/components/dialog/OpdsFeedAddForm.tsx:88:9
src/renderer/components/dialog/RenewLsdConfirm.tsx:51:9
src/renderer/components/dialog/ReturnLsdConfirm.tsx:52:9
src/renderer/components/dialog/SameFileImportConfirm.tsx:52:9
src/renderer/components/opds/BrowserResult.tsx:110:9
src/renderer/components/publication/TagManager.tsx:115:9
src/renderer/components/publication/menu/PublicationExportButton.tsx:61:9
src/renderer/components/reader/Reader.tsx:545:9
src/renderer/components/reader/Reader.tsx:651:17
src/renderer/components/reader/Reader.tsx:655:13
src/renderer/components/reader/UpdateBookmarkForm.tsx:66:9
src/renderer/components/searchResult/TextSearchResult.tsx:35:13
src/renderer/components/utils/SameFileImportManager.tsx:35:17
src/renderer/components/utils/SameFileImportManager.tsx:52:17
```

I think we should implement this correctly, by adding `async` and `await` where necessary (otherwise we are just obfuscating a coding mistake). For example:

```
    private async importFile(event: any) {
        const files = event.target.files;
        const paths: string[] = [];

        for (const f of files) {
            paths.push(f.path);
        }
        await this.props.importFiles(paths);
    }
```

...which in this case means that the `importFile()` function called by the `input@onChange` event handler will return a "dangling" / unhandled Promise, immediately discarded in the call stack (local variable in the function body), just like when `this.props.importFiles(paths)` is called without `await`.

```
onChange={ this.importFile }
```